trunk/source/Library-Automata/src/de/uni_freiburg/informatik/ultimate/automata/petrinet/unfolding/BranchingProcess.java

```diff
@@ -357,6 +323,38 @@ public boolean inCausalRelation(final Condition<LETTER, PLACE> condition, final
 		return eAncestors.contains(condition);
 	}
 
+	public boolean eventsInCausalRelation(final Event<LETTER, PLACE> e1, final Event<LETTER, PLACE> e2) {
+		final Set<Object> cAncestors = ancestorNodes(e1);
+		if (cAncestors.contains(e2)) {
+			return true;
+		}
+		final Set<Object> eAncestors = ancestorNodes(e2);
+		return eAncestors.contains(e1);
+	}
+
+	public boolean eventsInConcurrency(final Event<LETTER, PLACE> event1, final Event<LETTER, PLACE> event2) {
+		return (!eventsInConflict(event1, event2)) && (!eventsInCausalRelation(event1, event2));
+
+	}
+
+	public boolean eventsInConflict(final Event<LETTER, PLACE> e1, final Event<LETTER, PLACE> e2) {
+		if (e1 == e2) {
+			return false;
+		}
+		if (e1.getPredecessorConditions().stream().anyMatch(e2.getPredecessorConditions()::contains)) {
+			return true;
+		}
+		boolean result = false;
+		for (final Condition<LETTER, PLACE> e1Pred : e1.getPredecessorConditions()) {
+			for (final Condition<LETTER, PLACE> e2Pred : e2.getPredecessorConditions()) {
+				final Set<Object> e2Ancestors = ancestorNodes(e2);
+				result = result || conflictPathCheck(e1Pred, e2Pred, e2Ancestors);
+			}
+		}
+		return result;
+
+	}
+

```

Comment:
I assume this code is moved from somewhere else?

Reply:
No, this is just new code needed for Büchi-petri-nets.

Reply:
Ah, ok. I'm not sure though this is the best way to do it, though -- in particular I see the method `conflictPathCheck` here, which is (more efficient now than it was but still) not very efficient. Can you not use the `ICoRelation` provided by the `BranchingProcess` ?